src/Compilers/Core/Portable/Syntax/SyntaxNodeOrTokenList.cs

```diff
@@ -316,10 +316,10 @@ private static SyntaxNodeOrTokenList CreateList(List<SyntaxNodeOrToken> items)
                 return default(SyntaxNodeOrTokenList);
             }
 
-            var newGreen = GreenNode.CreateList(items.Select(n => n.UnderlyingNode!))!;
+            var newGreen = GreenNode.CreateList(items, static n => n.UnderlyingNode!)!;
```

Comment:
```suggestion
            var newGreen = GreenNode.CreateList(items, static n => n.RequiredUnderlyingNode)!;
```

Reply:
@HurricanKai you'll want to either undo this one, or add a `RequiredUnderlyingNode` to `SyntaxNodeOrToken` like `SyntaxToken` has.